infra/docker-ops.sh

```diff
@@ -1,42 +1,111 @@
 #!/bin/bash
 
-IMAGE_NAME=${1}
+# Input vars
+REPOSITORY_NAME=${1}
+FLAVOR_NAME=${2} # optional
 
-# shellcheck disable=SC2210
-IMAGE_VERSION=$(cat apps/"${IMAGE_NAME}"/VERSION 2>/dev/null || echo "${2}")
+# Static vars
+DOCKER_REPOS=("public.ecr.aws/htec")
+DIR="apps/${REPOSITORY_NAME}/${FLAVOR_NAME}"
+IMAGE_VERSION=$(cat "${DIR}"/VERSION 2>/dev/null || echo "${3}")
+IMAGES=()
 
-LOCAL_IMAGE="local/${IMAGE_NAME}:${IMAGE_VERSION}"
+# Main function is responsible for checking if manadatory variable is provided
+# If provided then loop through the docker repos and build an image for every repo
+main(){
+  if [ -z "$REPOSITORY_NAME" ] ; then
+      echo -e "     
+      #####################################\n
+      Mandatory argument is not provided\n
+      #####################################
+      "
+    else
+      for REPO in "${DOCKER_REPOS[@]}"; do
+        build_image $REPO $DIR
+      done
+  fi
+  # Wait for prepared data from previous function and proceed with the push and cleanup process.
+  push_image "$DRY_RUN"
+  remove_image "$DRY_RUN"
+}
 
-# We want to push same images on multiple docker registries
-DOCKER_REPOS=("public.ecr.aws/htec")
+build_image (){
+    local REPO=$1
+    local DIR=$2
+
+    DOCKERFILES=$(cd "$DIR"; find * -name '*Dockerfile')
+
+    if [ "$DRY_RUN" = "true" ] ; then
+          echo -e "     
+#####################################\n
+DRY_RUN is enabled, docker build command is:
+\n"
+      for i in ${DOCKERFILES[@]} ; do
```

Comment:
Why do you have redundant for-each loop. You should just skip docker build command, not to copy-paste the entire code.